src/PubSub/V1/Gapic/PublisherGapicClient.php

```diff
@@ -247,49 +247,26 @@ public static function parseName($formattedName, $template = null)
-     *     @type bool $forceNewChannel
-     *           If true, this forces gRPC to create a new channel instead of using a persistent channel.
-     *           Defaults to false.
-     *           NOTE: This option is only valid when utilizing the gRPC transport. Also, if the $channel
-     *           optional argument is specified, then this option is unused.
-     *     @type CredentialsLoader $credentialsLoader
-     *           A CredentialsLoader object created using the Google\Auth library.
+     *     Options for configuring the service API wrapper.
+     *
+     *     @type string $serviceAddress The address of the API remote host.
+     *           Default 'pubsub.googleapis.com:443'.
      *     @type string[] $scopes A string array of scopes to use when acquiring credentials.
-     *                          Defaults to the scopes for the Google Cloud Pub/Sub API.
+     *           Defaults to the scopes for the Google Cloud Pub/Sub API.
+     *     @type array $keyFile The contents of the service account credentials
+     *           .json file retrieved from the Google Developer's Console.
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
      *     @type string $clientConfigPath
      *           Path to a JSON file containing client method configuration, including retry settings.
      *           Specify this setting to specify the retry behavior of all methods on the client.
      *           By default this settings points to the default client config file, which is provided
-     *           in the resources folder. The retry settings provided in this option can be overridden
-     *           by settings in $retryingOverride
-     *     @type array $retryingOverride
-     *           An associative array in which the keys are method names (e.g. 'createFoo'), and
-     *           the values are retry settings to use for that method. The retry settings for each
-     *           method can be a {@see Google\ApiCore\RetrySettings} object, or an associative array
-     *           of retry settings parameters. See the documentation on {@see Google\ApiCore\RetrySettings}
-     *           for example usage. Passing a value of null is equivalent to a value of
-     *           ['retriesEnabled' => false]. Retry settings provided in this setting override the
-     *           settings in $clientConfigPath.
-     *     @type callable $authHttpHandler A handler used to deliver PSR-7 requests specifically
-     *           for authentication. Should match a signature of
-     *           `function (RequestInterface $request, array $options) : ResponseInterface`.
-     *     @type callable $httpHandler A handler used to deliver PSR-7 requests. Should match a
```

Comment:
All of the more advanced options have been removed, the rationale being what if we asked more advanced users to supply their own Transport implementation in such cases?

e.g.

```php
use Google\ApiCore\TransportFactory;
use Google\Cloud\PubSub\V1\PublisherClient;

$client = new PublisherClient([
    'transport' => TransportFactory::build([
        'httpHandler' => ...
    ])
]);

This keeps the base use case very simple and slims down the documentation, while still providing an avenue for advanced users to adjust nitty gritty details.

Reply:
+1

What do you think about having a `transportOptions` setting, such as:

`@type array $transportOptions An array of options to be passed to TransportFactory::build. Note that if the $transport option specifies an existing TransportInterface object, then this argument will be ignored.`

This supports an intermediate level of complexity. However, I can see an argument that this doesn't add enough value. WDYT?

Reply:
This makes sense, however, what if we used the name `advancedOptions` to avoid confusion with the fact we use `transportOptions` [elsewhere](https://github.com/googleapis/gax-php/blob/master/src/ApiCore/GapicClientTrait.php#L208).

Reply:
SGTM